# HG changeset patch
# User Vineet Kumar <vineet@doorstop.net>
# Date 1464576998 -3600
#      Mon May 30 03:56:38 2016 +0100
# Node ID fe311ed57a8df51a44e6d5bd7ea65b3389f9282d
# Parent  61fdfc04fa6cca48644fde747bb4ece129fd815b
[PATCH] pgp-verbose-mime
Configurable PGP mime part name patch

Orignal patch from http://www.doorstop.net/projects
---
 PATCHES   |  1 +
 globals.h |  2 ++
 init.h    | 12 ++++++++++++
 pgp.c     |  3 ++-
 4 files changed, 17 insertions(+), 1 deletion(-)

diff -r 61fdfc04fa6c -r fe311ed57a8d PATCHES
--- a/PATCHES	Mon May 30 03:46:36 2016 +0100
+++ b/PATCHES	Mon May 30 03:56:38 2016 +0100
@@ -1,3 +1,4 @@
+patch-vk.pgp_verbose_mime-gentoo
 patch-fg.change_folder_next-gentoo
 patch-vvv.quote-gentoo
 patch-xterm-no-tigetstr-on-tty-gentoo
diff -r 61fdfc04fa6c -r fe311ed57a8d globals.h
--- a/globals.h	Mon May 30 03:46:36 2016 +0100
+++ b/globals.h	Mon May 30 03:56:38 2016 +0100
@@ -275,6 +275,8 @@ WHERE REGEXP PgpDecryptionOkay;
 WHERE char *PgpSignAs;
 WHERE short PgpTimeout;
 WHERE char *PgpEntryFormat;
+WHERE char *PgpMimeSignatureFilename;
+WHERE char *PgpMimeSignatureDescription;
 WHERE char *PgpClearSignCommand;
 WHERE char *PgpDecodeCommand;
 WHERE char *PgpVerifyCommand;
diff -r 61fdfc04fa6c -r fe311ed57a8d init.h
--- a/init.h	Mon May 30 03:46:36 2016 +0100
+++ b/init.h	Mon May 30 03:56:38 2016 +0100
@@ -3333,6 +3333,18 @@ struct option_t MuttVars[] = {
   ** a line quoted text if it also matches $$smileys. This mostly
   ** happens at the beginning of a line.
   */
+  { "pgp_mime_signature_filename", DT_STR, R_NONE, UL &PgpMimeSignatureFilename, UL "signature.asc"},
+  /*
+  ** .pp
+  ** This option sets the filename used for signature parts in PGP/MIME
+  ** signed messages.
+  */
+  { "pgp_mime_signature_description", DT_STR, R_NONE, UL &PgpMimeSignatureDescription, UL "Digital signature"},
+  /*
+  ** .pp
+  ** This option sets the Content-Description used for signature parts in
+  ** PGP/MIME signed messages.
+  */
 
 
 
diff -r 61fdfc04fa6c -r fe311ed57a8d pgp.c
--- a/pgp.c	Mon May 30 03:46:36 2016 +0100
+++ b/pgp.c	Mon May 30 03:56:38 2016 +0100
@@ -1264,7 +1264,8 @@ BODY *pgp_sign_message (BODY *a)
   t->disposition = DISPNONE;
   t->encoding = ENC7BIT;
   t->unlink = 1; /* ok to remove this file after sending. */
-  mutt_set_parameter ("name", "signature.asc", &t->parameter);
+  mutt_set_parameter ("name", PgpMimeSignatureFilename, &t->parameter);
+  t->description = safe_strdup (PgpMimeSignatureDescription);
 
   return (a);
 }
